client/collection_field_description.go

```diff
@@ -43,6 +43,22 @@ type CollectionFieldDescription struct {
 	//
 	// This value has no effect on views.
 	DefaultValue any
+
+	// Size is a constraint that can be applied to fields that are arrays.
+	//
+	// Mutations on fields with a size constraint will fail if the size of the array
+	// does not match the constraint.
+	Size int
+
+	// Embedding contains the configuration for generating embedding vectors.
+	//
+	// This is only usable with array fields.
+	//
+	// When configured, embeddings may call 3rd party APIs inline with document mutations.
+	// This may cause increase latency in the completion of the mutation requests.
+	// This is necessary to ensure that the generated docID is representative of the
+	// content of the document.
+	Embedding *EmbeddingDescription
```

Comment:
suggestion: What if the structure/placement of the `EmbeddingDescription` was similar to how we have `IndexDescription`, where it actually lives on the `CollectionDescription` instead here on the `CollectionFieldDescription`.

This would make it much easier to iterate through the existing embedding descriptions when mutating documents to update/generate the embeddings. Whereas right now you have to iterate through all the fields on a doc. 

It also opens up the door to not having to actually have the embedding in the document graph itself, and instead is stored as an index outside the graph (future work).

Reply:
Done